internal/fwschemadata/value_semantic_equality_set.go

```diff
@@ -136,33 +140,44 @@ func ValueSemanticEqualitySetElements(ctx context.Context, req ValueSemanticEqua
 		// Ensure new value always contains all of proposed new value
 		newValueElements[idx] = proposedNewValueElement
 
-		if idx >= len(priorValueElements) {
+		if idx >= originalPriorElementsLength {
```

Comment:
Correctness: since we're handling the sets as unordered, I think this is not needed. A match for a prior set element could appear anywhere in the proposed set.